Basic/DyanmicPrefabs/Assets/Scripts/01_Server authoritative dynamic prefab loading across peers/DynamicPrefabManager.cs

```diff
@@ -0,0 +1,476 @@
+                response.Approved = false;
+                response.CreatePlayerObject = false;
+            }
+            
+            // In order for clients to not just get disconnected with no feedback, the server needs to tell the client why it disconnected it.
+            // This could happen after an auth check on a service or because of gameplay reasons (server full, wrong build version, etc)
+            // Since network objects haven't synced yet (still in the approval process), we need to send a custom message to clients, wait for
+            // UTP to update a frame and flush that message, then give our response to NetworkManager's connection approval process, with a denied approval.
+            IEnumerator WaitToDenyApproval(ulong clientID, ConnectStatus status, ICollection<AddressableGUID> addressableGUIDs)
+            {
+                response.Pending = true; // give some time for server to send connection status message to clients
+                response.Approved = false;
+                
+                m_DynamicPrefabGUIDs.Clear();
+                m_DynamicPrefabGUIDs.AddRange(m_LoadedDynamicPrefabResourceHandles.Keys);
+                SendServerToClientSetDisconnectReason(clientID, status, new AddressableGUIDCollection(){GUIDs = m_DynamicPrefabGUIDs.ToArray()});
+                yield return null; // wait a frame so UTP can flush it's messages on next update
+                response.Pending = false; // connection approval process can be finished.
+            }
+        }
+        
+        /// <summary>
+        /// This call attempts to spawn a prefab by it's addressable guid - it ensures that all the clients have loaded the prefab before spawning it,
+        /// and if the clients fail to acknowledge that they've loaded a prefab - the spawn will fail.
+        /// </summary>
+        /// <param name="guid"></param>
+        /// <returns></returns>
+        public async Task<(bool Success, NetworkObject Obj)> TrySpawnDynamicPrefabSynchronously(string guid, Vector3 position, Quaternion rotation)
+        {
+            if (IsServer)
+            {
+                var assetGuid = new AddressableGUID()
+                {
+                    Value = guid
+                };
+                
+                if (m_LoadedDynamicPrefabResourceHandles.ContainsKey(assetGuid))
+                {
+                    Debug.Log("Prefab is already loaded by all peers, we can spawn it immediately");
+                    var obj = await Spawn(assetGuid);
+                    return (true, obj);
+                }
+                
+                m_CountOfClientsThatLoadedThePrefab = 0;
+                m_SpawnTimeoutTimer = 0;
+                
+                Debug.Log("Loading dynamic prefab on the clients...");
+                LoadAddressableClientRpc(assetGuid);
+
+                int requiredAcknowledgementsCount = IsHost ? m_NetworkManager.ConnectedClients.Count - 1 : m_NetworkManager.ConnectedClients.Count;
```

Comment:
Added a fix for that - the server will now add the dynamic prefab before it will spawn it, thus any latejoiner will be required to download that prefab that's being synchrnously spawned. Good catch!